### PR TITLE
feat: Add hardhat mainnet forking

### LINKED ISSUE #537: Use hardhat fork feature for locahost development
We should use [Hardhat mainet forking feature](https://hardhat.org/guides/mainnet-forking.html) in order to enhance our dev workflow.
A PoC has been started here https://github.com/pancakeswap/pancake-frontend/pull/376.
Anyone with some Hardhat knowledge can pick up the work and set this up.

### PR DESCRIPTION
**Pull request by rickmort**
[ ] Before opening a pull request, please read the [contributing guidelines](https://github.com/pancakeswap/pancake-frontend/blob/master/CONTRIBUTING.md) first
[ ] If your PR is work in progress, open it as `draft`
[ ] Before requesting a review, all the checks need to pass
[ ] Explain what your PR does

fix #537

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * @type import('hardhat/config').HardhatUserConfig
+ */
+
+module.exports = {
+  solidity: '0.7.3',
+  networks: {
+    hardhat: {
+      forking: {
+        url: 'https://bsc-dataseed1.ninicoin.io',
+        blockNumber: 6771269,
```
Comment (hachiojidev) [inline hardhat.config.js:11]: Why a hard set block number?

Comment (rickmort) [reply hardhat.config.js:11]: In the [first attempt](https://github.com/pancakeswap/pancake-frontend/pull/376/commits/afa1e6fe5d981d754236b5c9755716b97d809e9b) to integrate Hardhat mainnet forking I saw an block number added.

Hardhat recommends branching from a specific block number to set up a fork-dependent test suite.

But it is completely optional.

what are your thoughts?

Comment (hachiojidev) [reply hardhat.config.js:11]: I can see why. I wonder though if it can be configured on a case-by-case basis.

**Comment by RabbitDoge** [comment]
Did you just copy the content from https://github.com/pancakeswap/pancake-frontend/pull/376 ?

**Pull request closed without merging**